hardware/arduino/avr/cores/arduino/USBCore.cpp

```diff
@@ -317,19 +306,19 @@ int USB_Send(u8 ep, const void* d, int len)
 	return r;
 }
 
-extern const u8 _initEndpoints[] PROGMEM;
-const u8 _initEndpoints[] = 
+u8 _initEndpoints[] =
 {
 	0,
 	
-#ifdef CDC_ENABLED
 	EP_TYPE_INTERRUPT_IN,		// CDC_ENDPOINT_ACM
 	EP_TYPE_BULK_OUT,			// CDC_ENDPOINT_OUT
 	EP_TYPE_BULK_IN,			// CDC_ENDPOINT_IN
-#endif
 
-#ifdef HID_ENABLED
-	EP_TYPE_INTERRUPT_IN		// HID_ENDPOINT_INT
+#ifdef PLUGGABLE_USB_ENABLED
+	//allocate 3 endpoints and remove const so they can be changed by the user
+	0,
+	0,
+	0,
 #endif
 };
 
```

Comment:
I dont know of there is a definition of the number of endpoints, but please consider that other MCUs dont have 6 endpoint and that not every endpoint can take double 64 bytes. If you want to make it flexible then let the use decide about the size and double bank or not.